### PR TITLE
gh-111968: Rename freelist related struct names to Eric's suggestion

### PR DESCRIPTION
**Pull request by corona10**
* Issue: gh-111968

### CODE REVIEW

**Comment by corona10** [comment]
https://github.com/python/cpython/issues/111968#issuecomment-1922178310

>  split `struct _Py_dictkeys_freelist` out of `struct _Py_dict_freelist`

Only one section is left :)

File: Include/internal/pycore_freelist.h [hunk 1]
```diff
@@ -109,20 +109,20 @@ struct _Py_async_gen_state {
...
+struct _Py_object_stack_freelist {
     struct _PyObjectStackChunk *free_list;
     Py_ssize_t numfree;
 };
 
-typedef struct _Py_freelist_state {
-    struct _Py_float_state floats;
-    struct _Py_tuple_state tuples;
-    struct _Py_list_state lists;
+typedef struct _PyObject_freelists {
```
Comment (colesbury) [inline Include/internal/pycore_freelist.h]: A few suggestions:

- I think the struct name (`_PyObject_freelists`) and typedef (`_PyFreeListState`) should  match. (Or get rid of the typedef or make the struct anonymous).
- The capitalization looks a bit weird to me. Maybe `_Py_object_freelists` or `_PyObject_Freelists`

File: Include/internal/pycore_tstate.h [hunk 2]
```diff
@@ -22,7 +22,7 @@ typedef struct _PyThreadStateImpl {
 
 #ifdef Py_GIL_DISABLED
     struct _mimalloc_thread_state mimalloc;
-    struct _Py_freelist_state freelist_state;
+    struct _PyObject_freelists freelist_state;
```
Comment (colesbury) [inline Include/internal/pycore_tstate.h]: `freelists` as the field name would be consistent with the per-interpreter definition.

File: Include/internal/pycore_freelist.h (hunk 1 above)
Comment (ericsnowcurrently) [reply Include/internal/pycore_freelist.h]: > * I think the struct name (`_PyObject_freelists`) and typedef (`_PyFreeListState`) should  match. (Or get rid of the typedef or make the struct anonymous).

I'd say drop the typedef.

> * The capitalization looks a bit weird to me. Maybe `_Py_object_freelists` or `_PyObject_Freelists`

+1 for `_Py_object_freelists` (or even  `_Py_object_freelists_t`).

**Comment by corona10** [comment]
CI failure is unrelated to this PR: see https://github.com/python/cpython/issues/115383

**Review by ericsnowcurrently — APPROVED** [review]
LGTM

**Review by colesbury — APPROVED** [review]
LGTM too!

**Pull request merged**